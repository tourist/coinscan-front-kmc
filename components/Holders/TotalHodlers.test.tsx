import { screen, render } from '@testing-library/react';
import snapshotDiff from 'snapshot-diff';
import TotalHodlers from './TotalHodlers';
import { testDataDailyHolders } from './test.fixture';

test('render total holders with loading pre-screen', async () => {
  const { asFragment, rerender } = render(<TotalHodlers loading />);
  const firstRender = asFragment();
  expect(firstRender).toMatchSnapshot('loading');
  rerender(
    <TotalHodlers
      loading={false}
      data={{ dailyHoldersStates: testDataDailyHolders }}
    />
  );
  await screen.findByText('6,341');
  await screen.findByText('0%');
  await screen.findByText('-0.14%');
  await screen.findByText('+0.89%');
  expect(snapshotDiff(firstRender, asFragment())).toMatchSnapshot('loaded');
});
