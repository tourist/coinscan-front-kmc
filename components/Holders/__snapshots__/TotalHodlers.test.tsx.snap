// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render total holders with loading pre-screen: loaded 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -8,40 +8,55 @@
        Total Holders
      </div>
      <div
        class=\\"MuiBox-root css-1n4i7xh\\"
      >
-       <span
-         class=\\"MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root\\"
-         style=\\"width: 150px; height: 70px;\\"
-       />
+       6,341
      </div>
      <div
        class=\\"MuiBox-root css-j7qwjs\\"
      >
        <div
          class=\\"MuiBox-root css-1xhj18k\\"
        >
-         <span
-           class=\\"MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root\\"
-           style=\\"width: 150px;\\"
-         />
+         <div
+           class=\\"MuiBox-root css-daxnmb\\"
+         >
+           0%
+         </div>
+         <div
+           class=\\"MuiBox-root css-1mvg5zt\\"
+         >
+           vs. last day
+         </div>
        </div>
        <div
          class=\\"MuiBox-root css-1xhj18k\\"
        >
-         <span
-           class=\\"MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root\\"
-           style=\\"width: 150px;\\"
-         />
+         <div
+           class=\\"MuiBox-root css-1d6tm0y\\"
+         >
+           -0.14%
+         </div>
+         <div
+           class=\\"MuiBox-root css-1mvg5zt\\"
+         >
+           vs. last 7 days
+         </div>
        </div>
        <div
          class=\\"MuiBox-root css-1xhj18k\\"
        >
-         <span
-           class=\\"MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root\\"
-           style=\\"width: 150px;\\"
-         />
+         <div
+           class=\\"MuiBox-root css-s7l08z\\"
+         >
+           +0.89%
+         </div>
+         <div
+           class=\\"MuiBox-root css-1mvg5zt\\"
+         >
+           vs. last 30 day
+         </div>
        </div>
      </div>
    </div>
  </DocumentFragment>"
`;

exports[`render total holders with loading pre-screen: loading 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiBox-root css-1jolfkj"
    >
      Total Holders
    </div>
    <div
      class="MuiBox-root css-1n4i7xh"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
        style="width: 150px; height: 70px;"
      />
    </div>
    <div
      class="MuiBox-root css-j7qwjs"
    >
      <div
        class="MuiBox-root css-1xhj18k"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
          style="width: 150px;"
        />
      </div>
      <div
        class="MuiBox-root css-1xhj18k"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
          style="width: 150px;"
        />
      </div>
      <div
        class="MuiBox-root css-1xhj18k"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
          style="width: 150px;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
